src/safeposix/syscalls/net_calls.rs

```diff
@@ -1399,23 +1639,51 @@ impl Cage {
+                                    // nonblocking I/O mode.  In nonblocking mode it would fail with
+                                    // the error EAGAIN or
+                                    // EWOULDBLOCK in this case.
                                     let mut nonblocking = false;
                                     if sockfdobj.flags & O_NONBLOCK != 0 {
                                         nonblocking = true;
                                     }
                                     let retval = match sockinfo.sendpipe.as_ref() {
+                                        //sendpipe is available in unix socket info
+                                        //it is needed to send the message found in buf
+                                        //to the connected socket
                                         Some(sendpipe) => {
                                             sendpipe.write_to_pipe(buf, buflen, nonblocking) as i32
                                         }
+                                        //sendpipe is not available in unix socket info
+                                        //transmission of data is not possible so return with error
                                         None => {
                                             return syscall_error(Errno::EAGAIN, "writev", "there is no data available right now, try again later");
                                         }
                                     };
-                                    if retval == -(Errno::EPIPE as i32) {
+                                    //In the case that the write_to_pipe call returns EPIPE,
+                                    // meaning the local end has been shut down on a connection
+                                    // oriented socket. Then, the process will also receive
+                                    // a SIGPIPE unless MSG_NOSIGNAL is set.
+                                    if (retval == -(Errno::EPIPE as i32))
+                                        && ((flags & MSG_NOSIGNAL) == 0)
+                                    {
+                                        // The default action for SIGPIPE is to terminate the
+                                        // process without a core dump. This simplifies error
+                                        // handling in programs that
+                                        // are meant to run as part of a shell pipeline: reading
+                                        // input, transforming it, and then writing it to another
+                                        // process. SIGPIPE allows the program to skip error
+                                        // handling and blindly write data until it’s killed
+                                        //
+                                        // Trigger SIGPIPE
                                         interface::lind_kill_from_id(self.cageid, SIGPIPE);
```

Comment:
add a bug tag here due to this [issue](https://github.com/Lind-Project/safeposix-rust/issues/306)

Reply:
+1